pythran/pythonic/include/types/nditerator.hpp

```diff
@@ -12,15 +12,9 @@ namespace pythonic
      */
     template <class E>
     struct nditerator
-        : public std::iterator<
-              std::random_access_iterator_tag,
-              typename std::remove_reference<decltype(
-                  std::declval<E &>().fast(0))>::type,
-              ptrdiff_t, typename std::remove_reference<decltype(
-                             std::declval<E &>().fast(0))>::type *,
-              typename std::remove_reference<decltype(std::declval<E &>().fast(
-                  0))>::type /* no ref here, because we return a proxy object */
-              > {
+        : public std::iterator<std::random_access_iterator_tag,
+                               typename std::remove_reference<decltype(
+                                   std::declval<E &>().fast(0))>::type> {
```

Comment:
On Wed, Dec 30, 2015 at 12:08:42PM -0800, pbrunet wrote:
> It will not be a problem that the field "reference" is empty? In case the std do something with : A::iterator_trait::reference or something like this?

Yeah the default is to use value_type &